### PR TITLE
Updated Readme file to contains docker desktop deployment guide

### PR DESCRIPTION
**Pull request by phoenix-henri**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
 To deploy Magento to different environments (develop, staging, production) it is recommended to create a `values_*.yaml`
 for each environment and tune the resource limits and configuration values of the services.
 
+## Docker Desktop Example
+The file `values_docker.yaml` is a shorter version of `values.yaml` which contains all the changes necessary
```
Comment (phoenix-bjoern) [inline README.md:170]: @phoenix-henri `values_docker.yaml` inherits all valeus from `values.yaml`. Therefore it only changes values necessary to run the Helm chart in the Docker Desktop K8S environment.

File: README.md [hunk 2]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
 To deploy Magento to different environments (develop, staging, production) it is recommended to create a `values_*.yaml`
 for each environment and tune the resource limits and configuration values of the services.
 
+## Docker Desktop Example
+The file `values_docker.yaml` is a shorter version of `values.yaml` which contains all the changes necessary 
+to execute the chart locally on your machine and create a local cluster with the help of [Docker Desktop](https://www.docker.com/products/docker-desktop).
```
Comment (phoenix-bjoern) [inline README.md:171]: @phoenix-henri Change "to execute" to "to deploy"

File: README.md [hunk 3]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+The following steps will give a guide on how to achieve that:
+
+### Prerequisites
+- Install *Docker Desktop* locally (Tested with 4.10.1) and enable kubernetes in settings.
```
Comment (phoenix-bjoern) [inline README.md:175]: @phoenix-henri "Install *Docker Desktop* >= 4.10 on your local machine and enable Kubernetes in the settings to start the cluster."

File: README.md [hunk 4]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+It is also recommended to increase the available resources of Docker Desktop to 4 CPU cores and 8 GB of RAM to ensure a smooth operation. 
+Alternatively you may install *kubectl* separately, however that may be a more error-prone approach.
+- Clone this repository locally on your computer.
+
+### Step 1
+Generate a `kube-config` file which will be used for the helm deployment. Make sure to generate it
```
Comment (phoenix-bjoern) [inline README.md:181]: @phoenix-henri I think this step can be skipped. If kubectl already "known" the local cluster, the command below will not provide a new connection other to ~/.kube/config. So this step is redundant. Maybe it is better to verify the kubectl context and reference https://docs.docker.com/desktop/kubernetes/#use-the-kubectl-command.

File: README.md [hunk 5]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+in the same folder as the git repo. Quickest way to do so is:
+
+```
+kubectl config view --raw > <name-of-your-file>
+```
+
+### Step 2
+Have a helm environment set up. We recommend using a docker container for that in order to facilitate the usage of Helm
```
Comment (phoenix-bjoern) [inline README.md:189]: @phoenix-henri Use consistent typing for "Helm" (not "helm") and "Docker" (not "docker"). Helm is not an environment, it is a package manager, a command line tool.

Be precise and less verbose in the README.

You can rephrase this sentence like "Install Helm. For ease of use it is recommended to run it in a Docker container.".

File: README.md [hunk 6]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+You may set up the docker container using the following command:
+
+```
+docker run -ti --entrypoint= -v $(pwd):/apps alpine/helm:latest sh
+```
+
+When using the native Helm installation bear in mind that the Helm version needs to be 3.9.x
+
+### Step 3
+Next we want to deploy the helm chart, but before we can do that we will need to pull the dependencies of our chart.
```
Comment (phoenix-bjoern) [inline README.md:200]: @phoenix-henri Sounds like you are talking to someone. Be descriptive.
"Step 3: Pull chart dependencies and deploy the Helm chart."

File: README.md [hunk 7]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+
+```
+helm dependency update #pulls all the other charts that our chart uses
+helm upgrade -i --kubeconfig <name-of-your-kubeconfig-flie> -f values_docker.yaml --create-namespace -n <your-namespace> magento .
+```
+
+### Step 3.5
+Check the progress of the deployment in the host machine, *not in your docker container*
```
Comment (phoenix-bjoern) [inline README.md:208]: @phoenix-henri "Watch the progress of the Helm deployment in Kubernetes"

File: README.md [hunk 8]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+
+```
+kubectl get pods -n <your-namespace> -w
+```
+
+### Step 4
+Once the deployment has fully finished all that is missing is an ingress-nginx, which we deploy through Helm as well.
```
Comment (phoenix-bjoern) [inline README.md:215]: @phoenix-henri "Docker Desktop ships without a Kubernetes Ingress (https://kubernetes.io/docs/concepts/services-networking/ingress/). To access the Magento store through your browser deploy an ingress by the following step:"

File: README.md [hunk 9]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+In the previously created docker container run the following command:
+
+```
+helm upgrade --kubeconfig kube-config --install ingress-nginx ingress-nginx --repo https://kubernetes.github.io/ingress-nginx --namespace ingress-nginx --create-namespace
```
Comment (phoenix-bjoern) [inline README.md:219]: @phoenix-henri Add the "--wait" parameter, so the cmd finishes when the ingress-nginx has been deployed succesfuly.

File: README.md [hunk 10]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+```
+
+The command above will deploy a ingress-nginx pod that has access to the `/etc/hosts file`.
```
Comment (phoenix-bjoern) [inline README.md:222]: @phoenix-henri "For the Magento installation the domain name "magento.local" is predefined in `values_docker.yaml`. Ensure the domain points to your local K8S cluster, so the Ingress can forward the traffic to the Magento instance:"

File: README.md [hunk 11]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+Now all we need to do is add an entry for our domain, which if unaltered is *magento.local*
+
+```
+echo "127.0.0.1 magento.local" >> /etc/hosts
+```
+
+### Step 5
+Navigate to `http://magento.local` in your browser and play around with your local magento installation!
```
Comment (phoenix-bjoern) [inline README.md:230]: @phoenix-henri Use consistent typing for "Magento" (not "magento")

File: README.md [hunk 12]
```diff
@@ -166,6 +166,91 @@ optional `--wait --timeout 15m` parameters in a deployment pipeline to see if th
...
+  env:
+    - name: MAGENTO_CLOUD_ROUTES
+      value: <base64-encoded-string-containing-your-domain>
+
+ingress:
+  hosts:
+    - name: <your-domain>
+```
+
+Easiest way to generate the base64 encoded strings is to use the ones we provide as a base.
```
Comment (phoenix-bjoern) [inline README.md:252]: @phoenix-henri well, this sentence redundant. Better reference the ece-tools section https://github.com/PHOENIX-MEDIA/magento2-helm#magento-ece-tools , which explains how to encode the string.

File: README.md (hunk 9 above)
Comment (phoenix-henri) [reply README.md:214]: it does that automatically in this case

**Pull request merged**